### DISCUSSION TITLE
Advanced config changes don't persist

### DISCUSSION THREAD

**Discussion by jscottelblein** [JulianPrieber/llc-docker#14] [Issues]
Using the latest version of LLC, I've noticed a few times now that changes I make to the Advanced config don't get saved between container restarts. Is there some file I need to mount so it persists?

**Comment by JulianPrieber** [comment]
I'm currently working on including data persistence in the documentation.

The easiest solution I came up with is moving the entire web application to a persistent volume.

**Something like this:**

1.  Create volume:
`docker volume create llc`

2.  Running the Docker image normally while mounting the volume:
`docker run --detach --name littlelink-custom --publish 80:80 --publish 443:443 --restart unless-stopped --mount source=llc,target=/htdocs julianprieber/littlelink-custom`

> **--mount source=llc,target=/htdocs julianprieber/littlelink-custom**

<br>

Let me know how this worked for you, and I'll include it in the docs 👍

**Comment by jscottelblein** [comment]
I don't know about the inner workings of this specific app, but I usually like to keep mounted things for only specific config (rather than entire apps where some or many of the files never change). There's already a database in use for LLC; is it possible to just save the changes (and use them) from a db table in it?

**Comment by JulianPrieber** [comment]
If you wish to save only selective files, you should save the following files and folders:

```
/htdocs/.env
/htdocs/database/database.sqlite
/htdocs/config/advanced-config.php
/htdocs/littlelink/images/avatar.png
/htdocs/themes (folder)
/htdocs/img (folder)
```

This might change with future releases.

**Comment by YouveGotMeowxy** [comment]
Is database.sqlite required when using a mysql db?

**Comment by JulianPrieber** [comment]
> Is database.sqlite required when using a mysql db?

In that case, it is not required.

**Comment by YouveGotMeowxy** [comment]
Last question (i think lol), what is `/htdocs/littlelink/images/avatar.png` for (like, where is it used) ? I'm confused because the user uploads their avatar in their profile and it gets saved in `img`, and also because it's an actual .png instead of an editable file, unless it is often deleted and re-saved as the same file name by LLC?

**Comment by JulianPrieber** [comment]
The avatar.png is used to store the custom site logo.

![chrome_xFp7lY1FID](https://user-images.githubusercontent.com/60265788/188115422-254c3500-acb7-4acf-9ef4-1f422536d8b5.png)

If you do not plan on using that one, you can skip that as well.

**Comment by YouveGotMeowxy** [comment]
> The avatar.png is used to store the custom site logo.
> 
> ![chrome_xFp7lY1FID](https://user-images.githubusercontent.com/60265788/188115422-254c3500-acb7-4acf-9ef4-1f422536d8b5.png)
> 
> If you do not plan on using that one, you can skip that as well.

ok thanks :)

just so I fully understand, if I change the logo, it deletes this one and renames the uploaded one to "avatar.png?" (and how come the logo is called an avatar? haha)

**Comment by JulianPrieber** [comment]
> just so I fully understand, if I change the logo, it deletes this one and renames the uploaded one to "avatar.png?" (and how come the logo is called an avatar? haha)

**Yup, pretty much like that.**

I can't really say for certain why the image is called that by I think I got a fairly good understanding why.

This function was implemented in the application LLC is forked from (https://github.com/khzg/littlelink-admin).
This thing is again based on the plain HTML project LittleLink (https://github.com/sethcottle/littlelink). In this thing, the profile image on the "home page" is called avatar.png in the files. So I think Khashayar just kept that name when implementing this.

We still even have the original comments from LittleLink in the source code, so this project was kept very close to the source material.

**Comment by JulianPrieber** [comment]
https://github.com/JulianPrieber/llc-docker/commit/0c72b576fa854ef9deebd2b78867c2f4ea751c41
